src/plugins/intel_gpu/include/intel_gpu/runtime/layout.hpp

```diff
@@ -192,48 +241,57 @@ struct padding {
+        }
+        return ret;
     }
 
     size_t hash() const {
         size_t seed = 0;
+        // std::cout << "=======" << __LINE__ << ": seed = " << seed << ", _filling_value = " << _filling_value << ", this = " << this << std::endl;
         seed = cldnn::hash_combine(seed, _filling_value);
-        seed = cldnn::hash_combine(seed, _lower_size.hash());
-        seed = cldnn::hash_combine(seed, _upper_size.hash());
-        seed = cldnn::hash_combine(seed, _dynamic_pad_dims.hash());
+        // std::cout << "=======" << __LINE__ << ": seed = " << seed << std::endl;
+        seed = cldnn::hash_combine(seed, hash_range(seed, std::begin(_lower_size), std::end(_lower_size)));
+        // std::cout << "=======" << __LINE__ << ": seed = " << seed << std::endl;
+        seed = cldnn::hash_combine(seed, hash_range(seed, std::begin(_upper_size), std::end(_upper_size)));
+        // std::cout << "=======" << __LINE__ << ": seed = " << seed << std::endl;
+        seed = cldnn::hash_combine(seed, hash_range(seed, std::begin(_dynamic_pad_dims), std::end(_dynamic_pad_dims)));
+        // std::cout << "=======" << __LINE__ << ": seed = " << seed << std::endl;
         return seed;
     }
 
     void save(BinaryOutputBuffer& ob) const {
-        ob << _lower_size.sizes();
-        ob << _upper_size.sizes();
+        std::vector<tensor::value_type> sizes;
+        ob << lower_size();
+        ob << upper_size();
         ob << _filling_value;
-        ob << _dynamic_pad_dims.sizes();
+        ob << get_dynamic_pad_dims();
     }
 
     void load(BinaryInputBuffer& ib) {
         std::vector<tensor::value_type> sizes;
         ib >> sizes;
-        _lower_size = tensor(sizes);
+        std::copy_n(sizes.begin(), sizes.size(), _lower_size);
         ib >> sizes;
-        _upper_size = tensor(sizes);
+        std::copy_n(sizes.begin(), sizes.size(), _upper_size);
         ib >> _filling_value;
         ib >> sizes;
-        _dynamic_pad_dims = tensor(sizes);
+        std::copy_n(sizes.begin(), sizes.size(), _dynamic_pad_dims);
     }
 
 private:
-    tensor _lower_size;  ///< Lower padding sizes. For spatials, it means size of left (X) and top (Y) padding.
-    tensor _upper_size;  ///< Upper padding sizes. For spatials, it means size of right (X) and bottom (Y) padding.
     // TODO: Add support for non-zero filling value (if necessary) or remove variable (if not necessary).
     float _filling_value;  ///< Filling value for an element of padding. If data type of elements is different than float it is converted
                            ///< to it using round-towards-nearest-even (for floating-point data types) or round-towards-zero (for integral
                            ///< data types).
-    tensor _dynamic_pad_dims = tensor(0);
+    tensor::value_type _lower_size[SHAPE_RANK_MAX] = {0};  ///< Lower padding sizes. For spatials, it means size of left (X) and top (Y) padding.
```

Comment:
Let's avoid using tensor::value_type as tensor will be removed eventually

Reply:
Sounds okay. Problem is which type is propriate? tensor::value_type is defined as "int". And from api layout::get_linear_offset(element), it sounds the input "element" could be of negative values. But ov::ParitialShape and ov::Shape uses size_t. To avoid type cast, it sounds size_t is better.

Reply:
done